test/functional/test-resources.js

```diff
@@ -1634,6 +1634,34 @@ describe('Resources discoverWork', () => {
       });
     });
   });
+
+  describe('passCallbacks_', () => {
+
+    it('should run passCallbacks_ added from onNextPass()', () => {
+      resources.isRuntimeOn_ = true;
+      resources.documentReady_ = true;
+      resources.firstPassAfterDocumentReady_ = true;
+
+      const passCallback = sandbox.spy();
+      resources.onNextPass(passCallback);
+      resources.doPass();
+
+      expect(passCallback).to.be.called;
+    });
+
+    it('should remove passCallbacks_ added from onNextPass(), ' +
+      ' after the pass', () => {
+      resources.isRuntimeOn_ = true;
+      resources.documentReady_ = true;
+      resources.firstPassAfterDocumentReady_ = true;
+
+      const passCallback = sandbox.spy();
+      resources.onNextPass(passCallback);
+      resources.doPass();
+
+      expect(resources.passCallbacks_.length).to.be.equal(0);
```

Comment:
Let's not write tests on `passCallbacks_`, since it's private. The intention here is that the `fn` isn't called twice, so let's run two passes and assert the `fn` isn't called twice.